packages/libs/client/src/contPact.ts

```diff
@@ -0,0 +1,197 @@
+
+    const command: IUnsignedCommand = {
+      hash,
+      sigs: this.sigs,
+      cmd,
+    };
+
+    this.cmd = command.cmd;
+    this.status = 'non-malleable';
+    return command;
+  }
+}
+
+/**
+ * Checks if the SPV Proof is successful or failed by polling the apiHost at
+ * a given interval. Times out if it takes too long.
+ *
+ * @param requestKey - the unique indentifier of the transaction
+ * @param targetChainId - the target chainweb of the transaction
+ * @param apiHost - the chainweb host where to send the transaction to
+ * @param options
+ *   - `interval` - the amount of time in ms between the api calls (optional)
+ *   - `timeout` - the total time in ms after this function will time out (optional)
+ *   - `onPoll` - `(status) => void` a function that gets called before each poll request (optional)
+ *
+ * @returns the SPV Proof Response
+ * @alpha
+ */
+export const pollSpvProof = async (
+  requestKey: string,
+  targetChainId: ChainId,
+  apiHost: string,
+  options?: {
+    interval?: number;
+    timeout?: number;
+    onPoll?: (status: string) => void;
+  },
+): Promise<Response | undefined> => {
+  const {
+    interval = 5000,
+    timeout = 1000 * 60 * 3,
+    onPoll = () => {},
+  } = { ...options };
+
+  const payload = {
+    targetChainId,
+    requestKey,
+  };
+
+  const startTime = Date.now();
+  let response;
+
+  while (true) {
+    try {
+      onPoll('Polling in progress');
+      response = await fetch(apiHost, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(payload),
+      });
+
+      if (response.status === 200) {
+        onPoll('Polling successful');
+        break;
+      }
+    } catch (error) {
+      console.error(error);
+      return response;
+    }
+
+    const elapsedTime = Date.now() - startTime;
+    if (elapsedTime >= timeout) {
+      onPoll('Timeout reached');
+      throw new Error('Timeout reached');
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, interval));
+  }
+
+  return response;
+};
+
+/**
+ * Creates a ContCommand instance with all the necessary .
+ * (i.e. it is checked whether the signatures are complete)
+ * @param requestKey - the unique indentifier of the transaction
+ * @param targetChainId - the target chainweb of the transaction
+ * @param apiHost - the chainweb host where to send the transaction to
+ * @returns the ContCommand instance with the set proof
+ * @alpha
+ */
+export async function getContCommand(
+  requestKey: string,
+  targetChainId: ChainId,
+  apiHost: string,
+): Promise<ContCommand> {
+  const proofResponse = await pollSpvProof(requestKey, targetChainId, apiHost, {
+    onPoll(status) {
+      console.log(status);
+    },
+  });
+
+  if (proofResponse === undefined) {
+    throw new Error('Unable to obtain SPV Proof');
+  }
+
+  const proof = await proofResponse.text();
+
+  const contCommand = new ContCommand(proof, 1, requestKey, false);
```

Comment:
I think the step could be something different, if the defpact has more than 2 steps , so maybe its better to accept it as an input